cpp/benchmarks/join/join_benchmark.cu

```diff
@@ -0,0 +1,87 @@
+/*
+ * Copyright (c) 2019, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include <benchmark/benchmark.h>
+#include <vector>
+#include <cudf/cudf.h>
+
+#include "generate_input_tables.cuh"
+
+
+template<typename key_type, typename payload_type>
+static void join_benchmark(benchmark::State& state)
+{
+    const gdf_size_type build_table_size {(gdf_size_type) state.range(0)};
+    const gdf_size_type probe_table_size {(gdf_size_type) state.range(1)};
+    const gdf_size_type rand_max_val {build_table_size * 2};
+    const double selectivity = 0.3;
+    const bool is_build_table_key_unique = true;
+
+    std::vector<gdf_column *> build_table;
```

Comment:
Use a `cudf::table`.

Reply:
I'm not sure this is a good idea. `cudf::table` constructor takes `std::vector<gdf_column*>` or an array of pointers. Since `gdf_inner_join` takes an array of pointers, we will need to get the pointer out of `cudf::table` we just created.